gips/data/landsat/landsat.py

```diff
@@ -286,6 +289,53 @@ def __init__(self, filename):
             raise Exception("Sensor %s not supported: %s" % (self.sensor, filename))
         self._version = self.version
 
+    def filter(self, pclouds=100, **kwargs):
+        if pclouds < 100:
+            if os.path.exists(self.filename):
+                mtlfilename = [f for f in self.datafiles() if 'MTL.txt' in f][0]
+                mtlfilename = self.extract([mtlfilename])[0]
+                with utils.error_handler('Error reading metadata file ' + mtlfilename):
+                    with open(mtlfilename, 'r') as mtlfile:
+                        text = mtlfile.read()
+                    cloud_cover = re.match(
+                        r".*CLOUD_COVER = (\d+.?\d*)",
+                        text,
+                        flags=re.DOTALL
+                    )
+                    if not cloud_cover:
+                        raise Exception("Error parsing MTL file")
```

Comment:
Optional:  This is a bit more precise:

```
cc_pattern = r".*CLOUD_COVER = (\d+.?\d*)" # set above re.match then call re.match(cc_pattern . . . )
if cloud_cover is None:
    # I can see leaving out mtlfilename here since you wisely put it in the error-handler above
    raise ValueError("No match for '{}' found in {}".format(cc_pattern, mtlfilename))
```



I like to imagine that I'm the one staring at the stack trace at 4:30 PM on a Friday, you know?  How can I make that guy's life easier?  : ]